src/tools/build-manifest/src/main.rs

```diff
@@ -293,6 +293,8 @@ impl Builder {
         self.write_channel_files(self.versions.channel(), &manifest);
         if self.versions.channel() != rust_version {
             self.write_channel_files(&rust_version, &manifest);
+            let major_minor = rust_version.split('.').take(2).collect::<Vec<_>>().join(".");
```

Comment:
`stable` is [not the only place where `self.versions.channel() != rust_version`](https://github.com/rust-lang/rust/blob/2033eb1495b4a0386a0ab08815769477baf20c89/src/tools/build-manifest/src/versions.rs#L230-L234): when you run `./x.py dist` locally you get a build for the `dev` channel, which has a `rust_version` of `1.XX.YY-dev`.

You should either gate this on the channel being `stable`, or handle the `dev` channel.